tools/c7n_azure/c7n_azure/resources/network_interface.py

```diff
@@ -29,3 +34,52 @@ class resource_type(object):
             'location',
             'resourceGroup'
         )
+
+
+@NetworkInterface.filter_registry.register('effective-route-table')
+class EffectiveRouteTableFilter(ValueFilter):
+    schema = type_schema('effective-route-table', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+        futures = []
+        results = []
+        chunk_size = 20
+
+        # Process each resource in a separate thread, returning all that pass filter
+        with self.executor_factory(max_workers=3) as w:
+            for resource_set in chunks(resources, chunk_size):
+                futures.append(w.submit(self.process_resource_set, resource_set))
+
+            for f in as_completed(futures):
+                if f.exception():
+                    self.log.warning(
+                        "Diagnostic settings filter error: %s" % f.exception())
+                    continue
+                else:
+                    results.extend(f.result())
+
+            return results
+
+    def process_resource_set(self, resources):
+        client = self.manager.get_client()
+        matched = []
+
+        for resource in resources:
+            try:
+                if 'routes' not in resource:
+                    route_table = (
+                        client.network_interfaces
+                            .get_effective_route_table(resource['resourceGroup'], resource['name'])
+                            .result()
+                    )
+
+                    resource['routes'] = route_table.serialize()
+                    filtered_effective_route_table = super(EffectiveRouteTableFilter, self).process([resource], event=None)

```

Comment:
You should be passing the event through, because in the event driven mode it could be relevant. I believe the ThreadHelper takes care of this though.